lib/validators/commit.js

```diff
@@ -0,0 +1,79 @@
+const { Validator } = require('./validator')
+const _ = require('lodash')
+const mustInclude = require('./options_processor/options/must_include')
+const constructOutput = require('./options_processor/options/lib/constructOutput')
+const consolidateResult = require('./options_processor/options/lib/consolidateResults')
+
+const MESSAGE_NOT_FOUND_ERROR = `Failed to run the 'commit' validator because 'message' option is not found. Please check README for more information about configuration`
+const REGEX_NOT_FOUND_ERROR = `Failed to run the test because 'regex' is not provided for 'message' option. Please check README for more information about configuration`
+const DEFAULT_FAIL_MESSAGE = `Some or all of your commit messages doesn't meet the criteria`
+const DEFAULT_SUCCESS_MESSAGE = `Your commit messages met the specified criteria`
+
+class Commit extends Validator {
+  constructor () {
+    super('dependent')
+    this.supportedEvents = [
+      'pull_request.*',
+      'pull_request_review.*'
```

Comment:
I feel we need to be more specific here and only specify `commit` events.

Reply:
sure, make sense, changed it to pull_request.synchronize only

Reply:
actually, that's not true because if we get receive other pull_request event, this validator will get skipped. I actually tested this and i was right. we need to keep `pull_request.*`

Reply:
If there are pull_request_reviews.* events would any of the commit change?
If I changed a label, milestones, description or title would any of the commit change?

Do we need to validate for the above events?

It seems redundant. Should we list on specific events where we know the commits will change? Why should commit validation happen for any other events when the commit sets have not changed?

Reply:
think about this scenario, 
- pull_request.opened, we check that the commit validation is failing, 
- now i add a label, and if we dont' check `commit` for that event, the same failing validation is now skipped, thus making it `pass`. I think your scenario would only work if we change store the state and change it's status but we are not doing that. we do a fresh check everytime.